### DISCUSSION TITLE
Custom created_at date on deposit/withdraw

### DISCUSSION THREAD

**Discussion by gera919** [bavix/laravel-wallet#619] [Q&A] [answered]
Does anyone have an idea how to overwrite the `created_at` during deposits or withdraw? I just want to put a custom date value during transaction creation

I tried to change the column value on the model boot `creating` hook but is not working at all for some reason.

Thanks!

**Comment by rez1dent3** [comment, marked as answer]
Hello. I would override the TransactionDtoTransformerInterface interface.

```php
use Bavix\Wallet\Internal\Dto\TransactionDtoInterface;
use Bavix\Wallet\Internal\Transform\TransactionDtoTransformer;
use Bavix\Wallet\Internal\Transform\TransactionDtoTransformerInterface;

final class TransactionDtoTransformerCustom implements TransactionDtoTransformerInterface
{
    public function __construct(
        private TransactionDtoTransformer $transactionDtoTransformer
    ) {
    }

    public function extract(TransactionDtoInterface $dto): array
    {
        return array_merge($this->transactionDtoTransformer->extract($dto), [
            'created_at' => new DateTimeImmutable($dto->getMeta()['created_at'] ?? 'now'),
        ]);
    }
}
```

And then in config/wallet.php:
```php
  //...
    'transformers' => [
        'transaction' => TransactionDtoTransformerCustom::class,
        //...
    ],
 //...
```

As a result, the composition can keep you backward compatible with the package in the future.

And now to use:
```php
$user->deposit(1000, ['created_at' => '2020-11-13 15:00:00']);
```

It will work.

**Reply by rez1dent3** [reply]
For simplicity, there is no validation and no checks.

**Reply by gera919** [reply]
Appreciate your quick response, this works like a charm. 
Thanks!
